xla/service/spmd/dot_handler.cc

```diff
@@ -663,8 +676,11 @@ std::optional<WindowedEinsumConfig> GetWindowedEinsumConfiguration(
 
   if (output_lhs_non_contracting_partitions == num_partitions &&
       output_sharding_transposed_to_match_lhs == lhs_sharding &&
-      rhs_shape_size >=
-          options.threshold_for_windowed_einsum_mib * 1024 * 1024 &&
+      should_enable_windowed_einsum_with_threshold(
+          options,
+          (ShapeUtil::ByteSizeOf(rhs->shape()) +
+           ShapeUtil::ByteSizeOf(lhs->shape())),
```

Comment:
In internal tests, I sometimes see `lhs` and `rhs` being nullptr due to `original_hlo` being nullptr.

I'm not sure in what cases `original_hlo` is nullptr. Maybe just skip creating a windowed einsum if total_bytes_windowed_einsum_threshold is set and these are nullptr.

Reply:
yea i think that makes sense. i changed the function take in lhs and rhs so we can do the null check there.